webapp/src/main/webapp/src/app/sender/archive/sender-archive.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from "@angular/core";
+import { BaseService } from "../../shared/base.service";
+import { Observable } from "rxjs";
+import { SenderSyncArchive } from "./sender-archive";
+import { SenderSyncArchiveCountAndItems } from "./sender-sync-archive-count-and-items";
+
+const RESOURCE_NAME = 'sender/archive';
+
+@Injectable({
+	providedIn: 'root'
+})
+export class SenderArchiveService extends BaseService<SenderSyncArchive> {
+
+	getArchiveCountAndItems(): Observable<SenderSyncArchiveCountAndItems> {
+		return this.getCountAndItems(RESOURCE_NAME);
+	}
+
+	getSyncArchivedByDate(startDate: any, endDate: any): Observable<SenderSyncArchiveCountAndItems> {
```

Comment:
Aren't these strings actually?

Reply:
These dates can be undefined  too that's why i put any as type **getSyncArchivedByDate** method

Reply:
Then in that case limit the types to string and undefined, typescript supports this i.e. multiple types for a single argument